### PR TITLE
Feature/wcon 75 provide spi for aem page context aware resource

### PR DESCRIPTION
**Pull request by rubnig**
Added a WCM CurrentPageConfigurationBindingsResourceDetectionStrategy to use the currentPage instead of the SlingHttpServletRequest resource

See also: https://issues.apache.org/jira/browse/SLING-8849

And the following Apache Sling pull requests:
- https://github.com/apache/sling-org-apache-sling-caconfig-spi/pull/1
- https://github.com/apache/sling-org-apache-sling-caconfig-impl/pull/3

### CODE REVIEW

**Comment by henrykuijpers** [comment]
> [ERROR] Failed to execute goal on project io.wcm.caconfig.extensions: Could not resolve dependencies for project io.wcm:io.wcm.caconfig.extensions:bundle:1.6.1-SNAPSHOT: Could not find artifact org.apache.sling:org.apache.sling.caconfig.spi:jar:1.4.0-SNAPSHOT in oss-snapshots (https://oss.sonatype.org/content/repositories/snapshots)

Any suggestion on fixing this build problem, @stefanseifert ?

**Comment by stefanseifert** [comment]
> Any suggestion on fixing this build problem, @stefanseifert ?

only the current develop branch of the sling models is published to the apache snapshots maven repository - but not those from branches in PRs (this would not work anyways if there are multiple PRs in parallel). build will work when the PRs get applied.

File: extensions/src/main/java/io/wcm/caconfig/extensions/persistence/impl/ToolsConfigPagePersistenceStrategy.java [hunk 1]
```diff
@@ -36,12 +36,12 @@
 import java.util.Map;
 import java.util.regex.Pattern;
 
-import org.apache.commons.collections.IteratorUtils;
-import org.apache.commons.collections.Predicate;
-import org.apache.commons.collections.PredicateUtils;
-import org.apache.commons.collections.Transformer;
-import org.apache.commons.collections.iterators.FilterIterator;
-import org.apache.commons.collections.iterators.TransformIterator;
+import org.apache.commons.collections4.IteratorUtils;
```
Comment (henrykuijpers) [inline extensions/src/main/java/io/wcm/caconfig/extensions/persistence/impl/ToolsConfigPagePersistenceStrategy.java]: These changes don't belong in this PR

**Comment by stefanseifert** [comment]
superseded by  https://github.com/wcm-io/wcm-io-caconfig/pull/39
